From 5b9a42821ec246ae4f81e8f13465da2709a10a23 Mon Sep 17 00:00:00 2001
From: ArenM <aren@peacevolution.org>
Date: Wed, 27 Apr 2022 23:57:38 -0400
Subject: [PATCH sxmo-utils] Systemd compatibility

* Arch enables pipewire, mmsd, and vvmd services by default. This
  removes the superd config files for them to avoid issues.

* Change superctl to systemctl --user in mmsd & vvmd config scripts
  since we're managing it with systemd.

* Remove the callaudiod service, callaudiod is intended to be started
  with dbus actcivation.
---
 configs/default_hooks/sxmo_hook_start.sh      | 21 ++-----------------
 configs/superd/services/callaudiod.service    |  9 --------
 configs/superd/services/mmsd.service          |  7 -------
 .../superd/services/pipewire-pulse.service    |  9 --------
 configs/superd/services/pipewire.service      |  8 -------
 configs/superd/services/vvmd.service          |  7 -------
 configs/superd/services/wireplumber.service   |  9 --------
 scripts/core/sxmo_mmsdconfig.sh               |  4 ++--
 scripts/core/sxmo_vvmdconfig.sh               |  4 ++--
 9 files changed, 6 insertions(+), 72 deletions(-)
 delete mode 100644 configs/superd/services/callaudiod.service
 delete mode 100644 configs/superd/services/mmsd.service
 delete mode 100644 configs/superd/services/pipewire-pulse.service
 delete mode 100644 configs/superd/services/pipewire.service
 delete mode 100644 configs/superd/services/vvmd.service
 delete mode 100644 configs/superd/services/wireplumber.service

diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index 3565f71..26bb39a 100644
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -48,14 +48,6 @@ case "$SXMO_WM" in
 		;;
 esac
 
-if [ -f "${SXMO_MMS_BASE_DIR:-"$HOME"/.mms/modemmanager}/mms" ]; then
-	superctl start mmsd
-fi
-
-if [ -f "${SXMO_VVM_BASE_DIR:-"$HOME"/.vvm/modemmanager}/vvm" ]; then
-	superctl start vvmd
-fi
-
 # Start the desktop widget (e.g. clock)
 superctl start sxmo_desktop_widget
 
@@ -76,17 +68,8 @@ superctl start sxmo_notificationmonitor
 # To setup initial lock state
 sxmo_hook_unlock.sh
 
-superctl start pipewire
-superctl start pipewire-pulse
-superctl start wireplumber
-
-(
-	sleep 5 # let some time to pipewire
-	superctl start callaudiod
-
-	# Turn on the dbus-monitors for modem-related tasks
-	sxmo_daemons.sh start modem_monitor sxmo_modemmonitor.sh
-) &
+# Turn on the dbus-monitors for modem-related tasks
+sxmo_daemons.sh start modem_monitor sxmo_modemmonitor.sh
 
 sxmo_migrate.sh state || sxmo_notify_user.sh --urgency=critical \
 	"Config needs migration" "$? file(s) in your sxmo configuration are out of date and disabled - using defaults until you migrate (run sxmo_migrate.sh)"
diff --git a/configs/superd/services/callaudiod.service b/configs/superd/services/callaudiod.service
deleted file mode 100644
index 2399ffd..0000000
--- a/configs/superd/services/callaudiod.service
+++ /dev/null
@@ -1,9 +0,0 @@
-[Unit]
-Description=call audio routing daemon
-After=pipewire-pulse
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/callaudiod
-ExecOnFailure=/usr/bin/sxmo_notify_user.sh "superd failed to start callaudiod"
diff --git a/configs/superd/services/mmsd.service b/configs/superd/services/mmsd.service
deleted file mode 100644
index 7096ddb..0000000
--- a/configs/superd/services/mmsd.service
+++ /dev/null
@@ -1,7 +0,0 @@
-[Unit]
-Description=daemon for receiving and managing GSM MMS messages
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/mmsdtng
diff --git a/configs/superd/services/pipewire-pulse.service b/configs/superd/services/pipewire-pulse.service
deleted file mode 100644
index 10d78a2..0000000
--- a/configs/superd/services/pipewire-pulse.service
+++ /dev/null
@@ -1,9 +0,0 @@
-[Unit]
-Description=low-level multimedia framework - capture and playback for both audio and video
-After=pipewire
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/pipewire -c pipewire-pulse.conf
-ExecOnFailure=/usr/bin/sxmo_notify_user.sh "superd failed to start pipewire-pulse"
diff --git a/configs/superd/services/pipewire.service b/configs/superd/services/pipewire.service
deleted file mode 100644
index 707047e..0000000
--- a/configs/superd/services/pipewire.service
+++ /dev/null
@@ -1,8 +0,0 @@
-[Unit]
-Description=low-level multimedia framework - capture and playback for both audio and video
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/pipewire
-ExecOnFailure=/usr/bin/sxmo_notify_user.sh "superd failed to start pipewire"
diff --git a/configs/superd/services/vvmd.service b/configs/superd/services/vvmd.service
deleted file mode 100644
index 33d01e1..0000000
--- a/configs/superd/services/vvmd.service
+++ /dev/null
@@ -1,7 +0,0 @@
-[Unit]
-Description=lower level daemon that retrieves Visual Voicemail
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/vvmd
diff --git a/configs/superd/services/wireplumber.service b/configs/superd/services/wireplumber.service
deleted file mode 100644
index e1a0602..0000000
--- a/configs/superd/services/wireplumber.service
+++ /dev/null
@@ -1,9 +0,0 @@
-[Unit]
-Description=Session / policy manager implementation for PipeWire
-After=pipewire
-
-[Service]
-Type=simple
-Restart=always
-ExecStart=/usr/bin/wireplumber
-ExecOnFailure=/usr/bin/sxmo_notify_user.sh "superd failed to start wireplumber"
diff --git a/scripts/core/sxmo_mmsdconfig.sh b/scripts/core/sxmo_mmsdconfig.sh
index 78d9378..586c389 100644
--- a/scripts/core/sxmo_mmsdconfig.sh
+++ b/scripts/core/sxmo_mmsdconfig.sh
@@ -75,10 +75,10 @@ newfile() {
 
 mkdir -p "$MMS_BASE_DIR"
 
-superctl stop mmsd
+systemctl --user stop mmsd-tng
 
 finish() {
-	superctl start mmsd
+	systemctl --user start mmsd-tng
 }
 trap 'finish' EXIT
 
diff --git a/scripts/core/sxmo_vvmdconfig.sh b/scripts/core/sxmo_vvmdconfig.sh
index ab14e4e..0ef7e95 100644
--- a/scripts/core/sxmo_vvmdconfig.sh
+++ b/scripts/core/sxmo_vvmdconfig.sh
@@ -65,10 +65,10 @@ newfile() {
 
 mkdir -p "$VVM_BASE_DIR"
 
-superctl stop vvmd
+systemctl --user stop vvmd
 
 finish() {
-	superctl start vvmd
+	systemctl --user start vvmd
 }
 trap 'finish' EXIT
 
-- 
2.37.3

